src/ws/ws-server.js

```diff
@@ -192,6 +239,39 @@ function participantIdentified(participantId, sessionId, role) {
     });
 }
 
+function puzzle(sessionId, input, expected) {
+    sendToSession(sessionId, ui.puzzle(input, expected));
+}
+
+function puzzleChanged(sessionId, puzzleIndex, puzzleName, timeLimit) {
+    sendToSession(sessionId, ui.puzzleChanged(puzzleIndex, puzzleName, timeLimit));
+}
+
+function roundCountdownChanged(sessionId, roundCountdown) {
+    sendToSession(sessionId, ui.roundCountdownChanged(roundCountdown));
+}
+
+function roundPhaseChanged(sessionId, roundPhase) {
+    sendToSession(sessionId, ui.roundCountdownChanged(roundPhase));
+}
+
+function sessionState(message) {
+    const { participantId, sessionId } = message;
+    const participant = hall.get(null, participantId, sessionId);
+    const participantIds = getScoreParticipantIds(message.score);
+
+    return Promise.all([
+        getProfiles([participantId]),
+        getProfiles(participantIds),
+    ]).then(([profile, scoreProfiles]) => {
+        const score = fillScoreWithProfiles(message.score, scoreProfiles, participantIds);
+
+        sendToParticipant(participant[0], ui.sessionState(profile.displayName, message.puzzleIndex, message.puzzleCount, message.puzzle, message.roundPhase, message.roundCountdown, message.startCountdown, message.playerInput, score));
```

Comment:
Of course, it is. But with such notation it shows explicitly what is the message. Not critical for me, but I'd leave it so.